staging_service/AutoDetectUtils.py

```diff
@@ -0,0 +1,35 @@
+from typing import Optional
+
+
+class AutoDetectUtils:
+    _FILE_EXTENSION_MAPPINGS = None  # expects to be set by config
+    _MAPPINGS = None  # expects to be set by config
+
+    # SUPPORTED_APPS = json.load(AVAILABLE_APPS_FP)
+    # EXTENSION_MAPPINGS = json.load(EXTENSION_MAPPINGS_FP)
+
+    @staticmethod
+    def determine_possible_importers(filename: str) -> Optional[list]:
+        """
+        Given a filename, come up with a reference to all possible apps
+        :param filename: The filename to find applicable apps for
+        :return: A list of mapping references, or None if not found
+        """
+        if "." in filename:
```

Comment:
what happens if we have multiple suffixes? (not sure if this is a real use case, but i think i recall us discussing it at some point)

Reply:
@leia-sefkin Not sure what you mean. Could you provide an example? The last suffix is the authoritative suffix

Reply:
`.tar.gz` would be a common example of multiple suffixes. Some people also use `.` as a separator in file names, e.g. `my.fave.seqs.10-10-2020.fna`

Reply:
The current solution would map both those examples correctly.